aiohttp/helpers.py

```diff
@@ -244,6 +243,31 @@ class ProxyInfo:
     proxy_auth: Optional[BasicAuth]
 
 
+def basicauth_from_netrc(netrc_obj: netrc.netrc, host: str) -> Optional[BasicAuth]:
+    auth_from_netrc = netrc_obj.authenticators(host)
+    if auth_from_netrc is not None:
+        login, account, password = auth_from_netrc
+        if not login:
+            # login is provided, we just use it as username.
+            # If login and account are both provided, only login is used
+            username = login
+        elif account:
+            # login is not provided, but account is
+            username = account
+        else:
+            # neither login nore account are provided, we use blank username
+            # this matches python 3.11 behavior
+            username = ""
+
+        if password is None:
+            # Password is not provided, we use blank string.
+            # this matches python 3.11 behavior
+            password = ""
+
+        auth = BasicAuth(username, password)
+        return auth
+
```

Comment:
```suggestion
    return None

```
If a function should return a value, then it should be explicit.

Reply:
In fact, would be even nicer to read if we just put:
```
if auth_from_netrc is None:
    return None
```
And then unindented the rest of the code.